dissect/target/plugins/general/network.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any, Iterator, Union
+
+from flow.record.fieldtypes.net import IPAddress
+
+from dissect.target.helpers.record import (
+    MacInterfaceRecord,
+    UnixInterfaceRecord,
+    WindowsInterfaceRecord,
+)
+from dissect.target.plugin import Plugin, export, internal
+
+InterfaceRecord = Union[UnixInterfaceRecord, WindowsInterfaceRecord, MacInterfaceRecord]
+
+
+class NetworkPlugin(Plugin):
+    __namespace__ = "network"
+
+    def check_compatible(self) -> None:
+        pass
+
+    @internal
+    def find_interfaces(self) -> list[InterfaceRecord]:
+        yield from ()
```

Comment:
This shouldn't be internal? And probably named something else, maybe just `_interfaces`, so that subclasses can just implement that and not have to re-export `interfaces`.

I think you got confused with a part in the ticket that references internal methods for looking up/finding interfaces by IP/MAC/CIDR, like we have methods to look up users by [SU]ID.